src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -339,8 +339,8 @@ private ClassWriter classFilePrologue() {
         ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_MAXS + ClassWriter.COMPUTE_FRAMES);
         setClassWriter(cw);
         cw.visit(Opcodes.V1_8, NOT_ACC_PUBLIC + Opcodes.ACC_FINAL + Opcodes.ACC_SUPER,
-                CLASS_PREFIX + className, null, INVOKER_SUPER_NAME, null);
-        cw.visitSource(SOURCE_PREFIX + className, null);
+                CLASS_PREFIX.concat(className), null, INVOKER_SUPER_NAME, null);
```

Comment:
I prefer to use the existing common pattern using `+` as I believe this gain is in the noise.

Reply:
The goal here was to remove/reduce allocations and excess calls - and using String.concat might have been excessive.

The bulk of the overhead is the potentially many `CLASS_PREFIX + className` calls, often via `className()`. Calculating this once and storing it in an instance field `prefixedClassName` is more profitable and help better disambiguates between `className` and `className()` - which sound like they should be the same but aren't.